src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -66,4 +80,105 @@ public String version() {
     public ConfigDef config() {
         return SplunkSinkConnectorConfig.conf();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues =
+            config.configValues()
+                .stream()
+                .collect(Collectors.toMap(
+                    ConfigValue::name,
+                    Function.identity()));
+
+        SplunkSinkConnectorConfig connectorConfig;
+        try {
+            connectorConfig = new SplunkSinkConnectorConfig(connectorConfigs);
+        } catch (Exception e) {
+            log.error("Validating configuration caught an exception", e);
+            return config;
+        }
+
+        validateAccess(connectorConfig, configValues);
+
+        return config;
+    }
+
+    /**
+     * We validate access by posting an empty payload to the Splunk endpoint.
+     *
+     * For a valid endpoint and a valid token, this returns a HTTP 400 Bad Request with the
+     * payload: {"text":"No data","code":5}
+     * For a valid endpoint and an invalid token, this returns a HTTP 403 Forbidden with the
```

Comment:
Is there a way we could validate the endpoint, i.e, valid token but invalid endpoint case?

Reply:
We can only index the data with both valid endpoint and valid token. So with an invalid endpoint, we would not be able to successfully post the data?

Reply:
I was asking if we could have a validation error thrown in case of valid token and invalid endpoint case.

Reply:
We only get a successful response with both together. There's another endpoint we could hit for endpoint but that ends up with another REST call with not much upside, this combination seemed more efficient.